skbase/base/_tagmanager.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding: utf-8 -*-
+"""Mixin class for flag and configuration settings management."""
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+__author__ = ["fkiraly"]
+__all__ = ["_FlagManager"]
+
+
+import inspect
+from copy import deepcopy
+
+
+class _FlagManager:
+    """Mixin class for flag and configuration settings management."""
+
+    @classmethod
+    def _get_class_flags(cls, flag_attr_name="_flags"):
+        """Get class flags from estimator class and all its parent classes.
+
+        Parameters
+        ----------
+        flag_attr_name : str, optional, default = "_flags"
+            name of the flag attribute that is read
+
+        Returns
+        -------
+        collected_flags : dict
```

Comment:
Per [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#returns) if there is only a single return, just the return type can be listed on first line of "returns" description.

Reply:
Can you take a look at the returns section of all the docstrings in this PR for conformity with this.